src/v/pandaproxy/sharded_client_cache.h

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2022 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+#include "config/broker_authn_endpoint.h"
+#include "config/rest_authn_endpoint.h"
+#include "pandaproxy/types.h"
+
+#include <seastar/core/sharded.hh>
+
+#include <vector>
+
+namespace pandaproxy {
+
+class kafka_client_cache;
+
+class sharded_client_cache {
```

Comment:
you may find that you can simplify things by:

1. moving the logic from `sharded_client_cache` into the non-sharded client cache object
2. removing `sharded_client_cache` completely
3. creating an `ss::sharded<client_cache>` wherever you need to (e.g. panda proxy server or application.cc)

i think it is partially a matter of preference, and other would argue that it might make testing easier. in any case, it is up to you. i just wanted to point out that if you saw this pattern elsewhere it doesn't necessarily mean you have to use it.

Reply:
I think I'll keep the sharded_client_cache because:
1. It separates the implementation for sharding from the implementation from caching
2. The sharding pattern remains consistent throughout our REST services (i.e., src/v/pandaproxy/*).